### PR TITLE
vagrant: run tests with Fedora 38

### PR DESCRIPTION
**Pull request by rst0git**
This pull request updates the version of vagrant from 2.2.19 to 2.3.7, and the base Fedora VM image from 37 to 38.

### CODE REVIEW

File: scripts/ci/vagrant.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ setup() {
 	# Tar up the git checkout to have vagrant rsync it to the VM
 	tar cf criu.tar ../../../criu
 	# Cirrus has problems with the following certificate.
-	wget --no-check-certificate https://releases.hashicorp.com/vagrant/${VAGRANT_VERSION}/vagrant_${VAGRANT_VERSION}_"$(uname -m)".deb -O /tmp/vagrant.deb && \
+	wget --no-check-certificate https://releases.hashicorp.com/vagrant/${VAGRANT_VERSION}/vagrant_${VAGRANT_VERSION}-1_"$(dpkg --print-architecture)".deb -O /tmp/vagrant.deb && \
```
Comment (adrianreber) [inline scripts/ci/vagrant.sh:22]: What is the reason for this change?

**Comment by adrianreber** [comment]
> The vagrant test fails. Pls investigate what is going on there.

It says no space left on device. @rst0git maybe you need to increase the vagrant disk size

File: scripts/ci/vagrant.sh (hunk 1 above)
Comment (rst0git) [reply scripts/ci/vagrant.sh:22]: It looks like they have changed the format of the URL:
https://releases.hashicorp.com/vagrant/2.3.6/

**Comment by rst0git** [comment]
> It says no space left on device. @rst0git maybe you need to increase the vagrant disk size

I tried increasing the disk size, but it fails with the same error. Here is the output of `df` from the vagrant VM in CI:
```
$ df -h
Filesystem      Size  Used Avail Use% Mounted on
devtmpfs        4.0M     0  4.0M   0% /dev
tmpfs           2.0G   84K  2.0G   1% /dev/shm
tmpfs           781M  644K  780M   1% /run
/dev/vda5        39G  2.3G   37G   6% /
tmpfs           2.0G     0  2.0G   0% /tmp
/dev/vda2       966M  107M  794M  12% /boot
/dev/vda3       100M   12M   89M  12% /boot/efi
/dev/vda5        39G  2.3G   37G   6% /home
tmpfs           391M     0  391M   0% /run/user/1000
```

**Comment by adrianreber** [comment]
> > It says no space left on device. @rst0git maybe you need to increase the vagrant disk size
> 
> I tried increasing the disk size, but it fails with the same error. Here is the output of `df` from the vagrant VM in CI:
> 
> ```
> $ df -h
> Filesystem      Size  Used Avail Use% Mounted on
> devtmpfs        4.0M     0  4.0M   0% /dev
> tmpfs           2.0G   84K  2.0G   1% /dev/shm
> tmpfs           781M  644K  780M   1% /run
> /dev/vda5        39G  2.3G   37G   6% /
> tmpfs           2.0G     0  2.0G   0% /tmp
> /dev/vda2       966M  107M  794M  12% /boot
> /dev/vda3       100M   12M   89M  12% /boot/efi
> /dev/vda5        39G  2.3G   37G   6% /home
> tmpfs           391M     0  391M   0% /run/user/1000
> ```

That is strange. Can you re-create this error locally using Vagrant?

**Comment by rst0git** [comment]
> Can you re-create this error locally using Vagrant?

Yes, the same error appears when running `make -C scripts/ci vagrant-fedora-rawhide` locally.

**Comment by avagin** [comment]
@adrianreber sorry, I clicked rebase-and-merge and only then noticed that @rst0git requested your review. Pls take a look and if here is something wrong, we will fix on top.
